File: Math/Number Theory/Prime Number/Goldbach Conjecture/goldbach_conjecture.py
def goldbach_conjecture(prime_list, num):
    if num % 2 != 0:
        return -1
    cnt = 0
    for prime in prime_list:
        if prime > num / 2:
            break
        if (num - prime) in prime_list:
            cnt += 1
    return cnt

File: Math/Number Theory/Prime Number/Goldbach Conjecture/goldbachConjecture.java
    private static int goldbachConjecture(ArrayList<Integer> primeList, int num) {
        if (num % 2 != 0) return -1;
        int cnt = 0;
        for (Integer prime : primeList) {
            if (prime > num / 2) break;
            if (primeList.contains(num - prime)) ++cnt;
        }
        return cnt;
    }
